src/Subscriber/CategoryView.php

```diff
@@ -46,16 +46,48 @@ public function onPageLoaded(NavigationPageLoadedEvent $event): void
         $path             = $this->getPath($category);
         $disableImmediate = safeGetByName($category->getCustomFields())(OmikronFactFinder::DISABLE_SEARCH_IMMEDIATE_CUSTOM_FIELD_NAME);
         $isHome           = $event->getRequest()->get('_route') === 'frontend.home.page';
+        $addParams = $this->prepareAddParams();
+        $communication = [
+            'search-immediate' => !$isHome && !$disableImmediate ? 'true' : 'false',
+            'category-page'    => $this->prepareCategoryPath($path),
+        ];
+
+        if ($addParams) {
+            $communication['add-params'] = $addParams;
+        }
 
         $event->getPage()->getExtension('factfinder')->assign(
             [
-                'communication' => [
-                    'search-immediate' => !$isHome && !$disableImmediate ? 'true' : 'false',
-                    'add-params'       => $path ? implode(',', $this->initial + [sprintf('filter=%s', urlencode($this->fieldName . ':' . $path))]) : '',
-                ],
+                'communication' => $communication,
             ]);
     }
 
+    private function prepareAddParams(): ?string
+    {
+        if (empty($this->initial)) {
```

Comment:
the add-params from $this->initial tend to be a full expressions like 
"navigation=true", so only thing we need to do is to implode them using ',' as a separator